rmgpy/data/kinetics/family.py

```diff
@@ -2162,7 +2166,9 @@ def assignWeightsToEntries(entryNestedList, weightedEntries, N = 1):
                                     re.sub(r"(?<=]) \+ (?!Average)", ",'",      # + sign between average and non-average
                                     re.sub(r"(?<!]) \+ (?!Average)", "','",     # + sign between non-averages
                                     comment)))))))
-
+            
+                evalCommentString = evalCommentString.replace("),",")',")
```

Comment:
Do you have an example of the type of comment that this deals with? Would it be possible to update the regex above instead of adding this line?

Reply:
I can't remember where I got the original failures, but this 'made up' comment replicates the parsing problem.  I just added parentheses around Csrad everywhere it showed up.  I think I missed one, but it shouldn't matter in the parsing.  

"""Average of [Average of [O2b;O_(Csrad)] + Average of [O_atom_triplet;O_(Csrad) + CH2_triplet;O_(Csrad)] + Average of [H_rad;O_(Csrad) + Average of [Ct_rad/Ct;O_(Csrad) from training reaction 0] + Average of [O_pri_rad;O_(Csrad) + Average of [O_rad/NonDeC;O_(Csrad) + O_rad/NonDeO;O_(Csrad)]] + Average of [Cd_pri_rad;O_(Csrad)] + Average of [CO_pri_rad;O_(Csrad)] + Average of [C_methyl;O_(Csrad) + Average of [C_rad/H2/Cs;O_Csrad + C_rad/H2/Cd;O_(Csrad) + C_rad/H2/O;O_(Csrad)] + Average of [C_rad/H/NonDeC;O_(Csrad)] + Average of [Average of [C_rad/Cs3;O_(Csrad)]]]]]
Estimated using template [Y_rad_birad_trirad_quadrad;O_(Csrad)] for rate rule [CH_quartet;O_(Csrad)]
Euclidian distance = 2.0
family: Disproportionation
"""

After processing it becomes:  
"[['O2b;O_(Csrad)'],['O_atom_triplet;O_(Csrad)','CH2_triplet;O_(Csrad)'],['H_rad;O_(Csrad),['Ct_rad/Ct;O_(Csrad) from training reaction 0'],['O_pri_rad;O_(Csrad),['O_rad/NonDeC;O_(Csrad)','O_rad/NonDeO;O_(Csrad)']],['Cd_pri_rad;O_(Csrad)'],['CO_pri_rad;O_(Csrad)'],['C_methyl;O_(Csrad),['C_rad/H2/Cs;O_Csrad','C_rad/H2/Cd;O_(Csrad)','C_rad/H2/O;O_(Csrad)'],['C_rad/H/NonDeC;O_(Csrad)'],[['C_rad/Cs3;O_(Csrad)']]]]]"

This can't be evaluated because where there should be "),'" there are "),".  It should be possible to adapt the regex, but I think it's a lot harder to do and harder to read than my fix.